### PR TITLE
Add test for the /notifications API

### PR DESCRIPTION
**Pull request by dbkr**

### CODE REVIEW

File: tests/61push/09_notifications_api.pl [hunk 1]
```diff
@@ -0,0 +1,72 @@
+test "Notifications can be viewed with GET /notifications",
+   requires => [ local_user_fixture( with_events => 0 ),
+		 local_user_fixture( with_events => 0 ),
```
Comment (leonerd) [inline tests/61push/09_notifications_api.pl]: Leading whitespace alignment

File: tests/61push/09_notifications_api.pl [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      })->then( sub {
+         my ( $body ) = @_;
+
+         log_if_fail("first /notifications response", $body);
+
+         assert_json_keys( $body, "notifications" );
+
+         my $notifs = $body->{notifications};
+
+         assert_json_keys( $notifs->[0], qw(room_id actions event read ts) );
```
Comment (leonerd) [inline tests/61push/09_notifications_api.pl]: Wants whitespace inside `qw( things here )` parens

File: tests/61push/09_notifications_api.pl [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+         # XXX: We can't assert that a key is present but null so can't test
```
Comment (leonerd) [inline tests/61push/09_notifications_api.pl]: You could

    assert_ok( exists $hash->{key}, "key in hash" );

File: tests/61push/09_notifications_api.pl [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+         matrix_advance_room_receipt( $user1, $room_id, "m.read" => $notif->{event}{event_id} );
+      })->then( sub {
+         do_request_json_for( $user1,
+            method  => "GET",
+            uri     => "/unstable/notifications",
+         );
+      })->then( sub {
+         my ( $body ) = @_;
+
+         log_if_fail("first /notifications response", $body);
```
Comment (leonerd) [inline tests/61push/09_notifications_api.pl]: Parens whitespace

**Comment by erikjohnston** [comment]
What's the status of this?

**Pull request merged**